content/articles/automatic-sms-verification-with-the-sms-retriever-api-in-android/index.md

```diff
@@ -0,0 +1,304 @@
+
+### SMS Verification process
+
+For mobile number verification, you need to implement the client side first. Afterwards, the server side, to complete the verification procedure. Usually, you send the user's phone number to the server performing verification. The server then sends an OTP code to the phone number. The SMS Retriever API starts listening for an SMS containing OTP code. After receiving the code, send it back to your server to complete the verification process.
+
+### Why use Automatic SMS Retriever API
+
+- Google abolished all apps using `CALL_LOG` and `READ_SMS` permissions. This is because they violated of users privacy. This led to removal of apps  using these permissions from play store in January 19th, 2021.
+- Users experience is smooth and almost effortless when using the app with this feature
+
+### Step 1. Create a new Android studio project
+
+![Create Project](/automatic-sms-verification-with-the-sms-retriever-api-in-android/create-project.png)
+
+
+### Step2: Put dependencies on build.gradle file
+
+We are going to use the following: 
+- Apache Commons - We are going to use this library to help us extract the code from the SMS message.
+- Google Play Services API - This library holds the SMS retrieval class
+- EventBus - To listen for received SMS from the SMS Retrieval API, we'll use a BroadcastReceiver. EventBus is a publisher/subscriber pattern library. We use it to communicate between our BroadcastReceiver and Activity classes.
+
+### Step 3: Setup the XML layout for our project
+
+We'll create an Edit Text in this section. This Edit text will display one-time code obtained from our SMS message.
+
+### Sending the mobile number to your server
+
+In this step, you have to get the user's phone number from the `EditText`. Send this code to your verification server which will send you the one-time code. Because I don't have a verification server, I'm not going to use that method in this article. I'm going to send the SMS from another phone. The SMS I'll send will include a four-digit code. This code will be extracted and displayed on the Edit text we made in activity main.xml. We will tackle this later in the article.
+
+Once you obtain the user's mobile number from the Edit text, send it to the server perfoming the verification. The server will send back a code that to the number that was sent to it. This now will have verified the phone number to your application. To perform SMS verification on a server, check out [SMS Verification on Server]. (https://developers.google.com/identity/sms-retriever/verify)
+
+### Step 4: Getting an instance of the SmsRetriverClient object
+
+We'll first create an instance of the SmsRetrieverClient object. This is followed by invoking its initSmsRetriever instance function and adding `onSuccessListener` and `onFailureListener` to the task. We wrap all these code in a function for later use.
+
+```
+private fun initSmsListener() {
```

Comment:
Did you forget something?
Oh, yes! You forgot to add the `language identifier` in your code snippets.